### PR TITLE
cli: add show-config command

### PR DESCRIPTION
**Pull request by kaniini**
Closes: #188

### CODE REVIEW

File: pkg/cli/show-config.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	var extraRepos []string
+
+	cmd := &cobra.Command{
+		Use:   "show-config",
+		Short: "Show the configuration derived from loading a YAML file",
+		Long: `Show the configuration derived from loading a YAML file.
+
+The derived configuration is rendered in YAML.
+`,
+		Example: `  apko show-config <config.yaml>`,
```
Comment (imjasonh) [inline pkg/cli/show-config.go:40]:
```suggestion
		Example: "apko show-config <config.yaml>",
```

Unless there's a reason for it to have spaces prefixed?

File: pkg/cli/show-config.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&extraKeys, "keyring-append", "k", []string{}, "path to extra keys to include in the keyring")
+	cmd.Flags().StringSliceVarP(&extraRepos, "repository-append", "r", []string{}, "path to extra repositories to include")
+
+	return cmd
+}
+
+func ShowConfigCmd(ctx context.Context, opts ...build.Option) error {
```
Comment (imjasonh) [inline pkg/cli/show-config.go:58]: Can this be unexported? Or is this something we want to export so we can use it outside of apko?

File: pkg/cli/show-config.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	var buf bytes.Buffer
+	enc := yaml.NewEncoder(&buf)
+
+	if err := enc.Encode(bc.ImageConfiguration); err != nil {
+		return fmt.Errorf("failed to encode YAML document: %w", err)
+	}
+
+	if _, err := buf.WriteTo(os.Stdout); err != nil {
```
Comment (imjasonh) [inline pkg/cli/show-config.go:81]: Any reason not to write it straight to stdout? `yaml.NewEncoder(os.Stdout).Encode(bc.ImageConfiguration)`

(Obviously not a huge deal to buffer it, just curious)

**Review by imjasonh — APPROVED** [review]

File: pkg/cli/show-config.go (hunk 2 above)
Comment (kaniini) [reply pkg/cli/show-config.go:58]: Yes, it can be.

File: pkg/cli/show-config.go (hunk 3 above)
Comment (kaniini) [reply pkg/cli/show-config.go:81]: I wanted to catch any errors prior to dumping a partial doc, basically.

File: pkg/cli/show-config.go (hunk 1 above)
Comment (kaniini) [reply pkg/cli/show-config.go:40]: That's how cosign did it, I copied cosign :P

**Pull request merged**